### PR TITLE
Accept portrait imagery/video as input

### PR DESCRIPTION
**Pull request by MikiyaShibuya**
Fix the crash when the portrait images or videos are feed.

When any portrait images are given as input, the orb feature extractor made crashes due to only landscape imagery input are assumed. 
In order to fix the problem, I make some additional processes to accept the portrait images or videos.

### CODE REVIEW

File: src/openvslam/feature/orb_extractor.cc [hunk 1]
```diff
@@ -474,10 +474,30 @@ std::vector<cv::KeyPoint> orb_extractor::distribute_keypoints_via_tree(const std
...
+        num_y_grid = 1;
+        delta_x = static_cast<double>(max_x - min_x) / num_x_grid;
+        delta_y = max_y - min_y;
+    }
+    else {
+        // If the aspect ratio is less than 1, the patches are made in a vertical direction
+        num_x_grid = 1;
+        num_y_grid = std::round(1 / ratio);
+        delta_x = max_x - min_y;
+        delta_y = static_cast<double>(max_y - min_y) / num_y_grid;
```
Comment (shinsumicco) [inline src/openvslam/feature/orb_extractor.cc]: This scope seems to be processed when the aspect ratio is **equal to** or less than 1.
Could you modify the comment?

File: src/openvslam/feature/orb_extractor.cc [hunk 2]
```diff
@@ -500,7 +524,12 @@ std::list<orb_extractor_node> orb_extractor::initialize_nodes(const std::vector<
 
     // Assign all keypoints to initial nodes which own keypoint's position
     for (const auto& keypt : keypts_to_distribute) {
-        initial_nodes.at(keypt.pt.x / delta_x)->keypts_.push_back(keypt);
+        // x / y index of the patch where the keypt place
```
Comment (shinsumicco) [inline src/openvslam/feature/orb_extractor.cc]: "the keypt place" -> "the keypt is placed" ?

**Comment by fangxu622** [comment]
hi ，I changed the code according to your change, but it still runs failed on portrait imagery/video

File: src/openvslam/feature/orb_extractor.cc (hunk 2 above)
Comment (fangxu622) [reply src/openvslam/feature/orb_extractor.cc]: I still run  failed

**Comment by shinsumicco** [comment]
@fangxu622 
Could you tell us the more details?
(error message, stack trace, etc.)

@MikiyaShibuya 
Is there something for this?

**Comment by fangxu622** [comment]
> @fangxu622
> Could you tell us the more details?
> (error message, stack trace, image size, etc.)
> 
> @MikiyaShibuya
> Is there something for this?

I‘m sorry for late., the  error message like this , when the portrait imagery appears on the video

**Comment by shinsumicco** [comment]
@fangxu622 
Thank you for posting the error message.
The image you uploaded is landscape (cols > rows), isn't it?
The error message seems to be caused by not the feature extractor but the homography solver.
Could you create the new issue about it?

**Comment by shinsumicco** [comment]
According to #124, I conclude that the problem the feature extractor could not accept portrait imagery has been solved.
Thanks @MikiyaShibuya and @pouyaAB.

**Comment by fangxu622** [comment]
> @fangxu622
> Thank you for posting the error message.
> The image you uploaded is landscape (cols > rows), isn't it?
> The error message seems to be caused by not the feature extractor but the homography solver.
> Could you create a new issue about it?

@shinsumicco @MikiyaShibuya  Hi , Sorry ,  My video data have some bug that was resulted in this bug.  It's my problem

**Pull request merged**